build_scripts/windows/refactor.xslt

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0"?>
+<!-- Define WiX namespace twice: template matches and selections don't work with default namespace, but default handy for simply emitting elements. -->
+<xsl:stylesheet version="1.0" xmlns="http://schemas.microsoft.com/wix/2006/wi" xmlns:wix="http://schemas.microsoft.com/wix/2006/wi" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
```

Comment:
I didn't see anything specific to Azure CLI in `refactor.xslt`. Since it is beneficial to all MSI packages, is it a good idea to include it in the WiX Toolset itself?

Reply:
Not sure what you mean. Generally, 1 file per component is recommended but not when you have thousands of files/components. That's not common with MSIs, and optimizations like this - though, perhaps more targeted, which isn't really possible when harvesting - are common for MSIs.

As to why it took almost as long, I'd need to see a verbose log. One area of concern was that because of the component refactoring, I had to move `RemoveExistingProducts` earlier which means all the files are removed first then new ones added. If files weren't updated, there's still a larger IO hit. Still, if you most often update most files, this avoids unnecessary time doing CRC checks to see if a file changed.

One idea I had was to conditionally schedule `RemoveExistingProducts` early for older MSI versions (before the refactoring) but later for newer versions (after the refactoring), but that would require a custom action to do so. The first upgrade to the refactored MSI might be longer (still, I'm surprised you didn't see a positive delta) but subsequent ones would be faster.